types/node-pixel/index.d.ts

```diff
@@ -0,0 +1,52 @@
+import { Board } from "johnny-five";
+export {};
+
+// Opaque types typings
+declare const tag: unique symbol;
+interface TagContainer<Token> {
+    readonly [tag]: Token;
+}
+type Opaque<Type, Token = unknown> = Type & TagContainer<Token>;
+
+// Lib typings
+export const BACKWARD: Opaque<number, "ShiftDirection">;
+export const FORWARD: Opaque<number, "ShiftDirection">;
+export type Direction = typeof BACKWARD | typeof FORWARD;
+
+export namespace COLOR_ORDER {
+    const GRB: Opaque<number, "ColorOrder">;
+    const RGB: Opaque<number, "ColorOrder">;
+    const BRG: Opaque<number, "ColorOrder">;
+}
+
+export interface StripArgs {
+    board: Board;
+    strips: Array<{ pin: string | number; length: number }> | number[];
+    controller?: "FIRMATA" | "I2CBACKPACK";
+    color_order?: typeof COLOR_ORDER.GRB | typeof COLOR_ORDER.RGB | typeof COLOR_ORDER.BRG;
+    gamma?: number;
+}
+
+export class Strip {
+    readonly length: number;
+
+    constructor(args: StripArgs);
+
+    pixel(index: number): Pixel;
+    shift(index: number, direction: Direction, wrap: boolean): void;
+    color(color: string | number[]): void;
+    colour(color: string | number[]): void;
+    off(): void;
+    clear(): void;
+    show(): void;
+
+    on(event: string, callback: () => void): void;
+    on(event: "error", callback: (e: Error) => void): void;
+}
+
+export class Pixel {
+    color(color: string | number[]): void;
+    colour(color: string | number[]): void;
+    off(): void;
+    clear(): void;
+}
```

Comment:
```suggestion
```
was this symbol exported in source?
https://github.com/ajfisher/node-pixel/blob/master/lib/index.js

Reply:
No but could i do this using an exported interface? This way we will still be able to type the content of a pixel if we want it to be in a variable ?

```
export interface Pixel {
    color(color: string | number[]): void;
    colour(color: string | number[]): void;
    off(): void;
    clear(): void;
}
```

Reply:
Also this is used in Strip.pixel `pixel(index: number): Pixel;`

Reply:
@peterblazejewicz Used an exported interface so we can still type and this is correct.